src/ts/generic/featurecollection.ts

```diff
@@ -181,9 +181,25 @@ export function buildHeader(header: HeaderMeta, crsCode = 0): Uint8Array {
     Header.addName(builder, nameOffset);
     const offset = Header.endHeader(builder);
     builder.finishSizePrefixed(offset);
+
     return builder.asUint8Array() as Uint8Array;
 }
 
+export async function readMetadata(url: string, nocache = false, headers: HeadersInit = {}): Promise<HeaderMeta> {
+    const assumedHeaderLength = 2024;
```

Comment:
This assumption will not always work to get the complete header, it depends on how large schema (number of columns and names) are described by the header and other contents of the header metadata.

The true header length should be parsed and if larger than the assumed length it needs to make an additional request to get the rest.

Reply:
I don't think I have test data at hand to excercise it and make the correct fix with test case... so it's not entirely trivial to fix. I might be willing to accept this PR anyway with the known caveat that it could fail on some data. @stefanocudini what is your opinion?